src/test/java/org/opentripplanner/updater/stoptime/TimetableSnapshotSourceTest.java

```diff
@@ -903,25 +1030,65 @@ public void testHandleScheduledTripWithSkippedAndScheduled() {
     }
   }
 
-  @Test
-  public void testPurgeExpiredData() throws InvalidProtocolBufferException {
+  enum SameAssert {
+    Same {
+      public void doAssert(Object a, Object b) {
+        assertSame(a, b);
+      }
+    },
+    NotSame {
+      public void doAssert(Object a, Object b) {
+        assertNotSame(a, b);
+      }
+    };
+
+    public abstract void doAssert(Object a, Object b);
+  }
+
+  static Stream<Arguments> purgeExpiredDataTestCases = Stream.of(
+    Arguments.of(Boolean.TRUE, -1, NotSame, Same, Same, NotSame, NotSame),
+    Arguments.of(Boolean.FALSE, -1, NotSame, Same, Same, NotSame, NotSame),
+    Arguments.of(Boolean.TRUE, 1000, Same, Same, Same, NotSame, Same),
+    Arguments.of(Boolean.FALSE, 1000, Same, Same, Same, NotSame, Same)
+  );
```

Comment:
Changing the `purgeExpiredData` does not have an effect - the test should be changed so it test this parameter.